### DISCUSSION TITLE
creating matrix problem

### DISCUSSION THREAD

**Discussion by azzezzanf** [teamalgoritma/community#472] [Q&A] [answered]
i have a problem here, i dont really understand because its been around one and half years since i use R languange, can u pls help me?
<img width="1020" alt="Screen Shot 2022-12-05 at 13 52 52" src="https://user-images.githubusercontent.com/80685182/205570771-32daa38a-aeab-4171-9ad5-905a3632b167.png">
![question1](https://user-images.githubusercontent.com/80685182/205570848-45
<img width="1098" alt="Screen Shot 2022-12-05 at 14 02 38" src="https://user-images.githubusercontent.com/80685182/205571010-5d6ab5c7-7391-49f1-bca8-717aad6570ab.png">
98bbc4-6bd0-4aa7-8970-95ea6a83aa5c.png)

**Comment by dwiknrd** [comment, marked as answer]
Hi [azzezzanf](https://github.com/azzezzanf),


Actually I don't know exactly what you are doing and what you want to do. But hopefully this answer can help as a reference.

To create a matrics you can using the `matrix()` function. The syntax of the `matrix()` function is:

`matrix(data,byrow,nrow,ncol,dimnames)`

The arguments in the matrix function are the following:

- `data` – data contains the elements in the R matrix.
- `byrow` – byrow is a logical variable. Matrices are by default column-wise. By setting byrow as TRUE, we can arrange the data row-wise in the matrix.
- `nrow` – defines the number of rows in the R matrix.
- `ncol` – defines the number of columns in the R matrix.
- `dimnames` – takes two character arrays as input for row names and column names.

Notes: Make sure the amount of data matches the number of rows and columns created, so that no errors or warnings occur.

Next, if you want to create a custom function with argument, information or data can be passed into function as arguments. Arguments are specified after the function name, inside the parentheses. You can add as many arguments as you want, just separate them with a comma.
Here the example of creating function using 2 argument

```
my_function <- function(fname, lname) {
  paste(fname, lname)
}

my_function("Peter", "Griffin")
```

Please take a look the way that function called, we can passed information or data as argument in the function. When viewed from the code that you attached above, it will produce an error because the function that you created requires several arguments to produce output.

I hope this helps, cheers :)
